### PR TITLE
Patmos platform added

### PR DESCRIPTION
**Pull request by EhsanKhodadad**
In this commit, the support files for patmos processor platform is added.

### CODE REVIEW

File: core/platform/lf_patmos_support.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * TODO: For improved power consumption this should be implemented with a HW timer and interrupts.
+ *
+ * @param wakeup int64_t time of wakeup
+ * @return int 0 if successful sleep, -1 if awoken by async event
+ */
+
+int _lf_interruptable_sleep_until_locked(environment_t* env, instant_t wakeup) {
+    instant_t now;
+    _lf_async_event = false;
+    lf_disable_interrupts_nested();
```
Comment (erlingrj) [inline core/platform/lf_patmos_support.c:21]: Here interrupts should be enabled, not disabled. I think this is copied from the arduino support? I found the same bug there

File: core/platform/lf_patmos_support.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * This returns 0 (it never fails, assuming the argument gives a valid memory location).
+ * This has to be called at least once per 35 minutes to properly handle overflows of the 32-bit clock.
+ * TODO: This is only addressable by setting up interrupts on a timer peripheral to occur at wrap.
+ */
+
+int _lf_clock_gettime(instant_t* t) {
+
+    assert(t != NULL);
+
+   *t = get_cpu_usecs() * 1000;
```
Comment (erlingrj) [inline core/platform/lf_patmos_support.c:56]: What is the return type of `get_cpu_usecs`? Also, is it monotonic?

File: core/platform/lf_patmos_support.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    return 0;
+}
+
+#if defined(LF_SINGLE_THREADED)
+
+int lf_enable_interrupts_nested() {
+    // unmask interrupts
+    intr_unmask_all();
+    // clear pending flags
+    intr_clear_all_pending();
```
Comment (erlingrj) [inline core/platform/lf_patmos_support.c:67]: Are you sure you would like to clear pending interrupts? If there was an interrupt request while we were in a critical section we would like to jump to that interrupt handler when we enable interrupts again

File: core/platform/lf_patmos_support.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+   *t = get_cpu_usecs() * 1000;
+   
+    return 0;
+}
+
+#if defined(LF_SINGLE_THREADED)
+
+int lf_enable_interrupts_nested() {
+    // unmask interrupts
+    intr_unmask_all();
```
Comment (erlingrj) [inline core/platform/lf_patmos_support.c:65]: What does it mean to unmask the interrupts?

File: include/core/platform/lf_patmos_support.h [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#include <time.h>   // For CLOCK_MONOTONIC
+#include <stdbool.h>
+
+#include <inttypes.h>  // Needed to define PRId64 and PRIu32
+#define PRINTF_TIME "%" PRId64
+#define PRINTF_MICROSTEP "%" PRIu32
+#define PRINTF_TAG "(%" PRId64 ", %" PRIu32 ")"
+
+// Keep track of physical actions being entered into the system
+static volatile bool _lf_async_event = false;
```
Comment (erlingrj) [inline include/core/platform/lf_patmos_support.h:55]: I dont think you want this variable definition here?

File: include/core/platform/lf_patmos_support.h [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+// The underlying physical clock for Linux
+#define _LF_CLOCK CLOCK_MONOTONIC
```
Comment (erlingrj) [inline include/core/platform/lf_patmos_support.h:58]: No need to define this

**Review by erlingrj — CHANGES_REQUESTED** [review]
This looks promising. I have some requests and some clarifying questions only :)

File: core/platform/lf_patmos_support.c (hunk 2 above)
Comment (EhsanKhodadad) [reply core/platform/lf_patmos_support.c:56]: The return value is unsigned long long, and it's monotonically increasing.

**Comment by lhstrh** [comment]
@EhsanKhodadad, in the interested of this PR not falling to far behind, would you be able to address @erlingrj's feedback at your earliest convenience?

**Comment by lhstrh** [comment]
Great! Thanks, @EhsanKhodadad.

**Pull request closed without merging**